docs/pages/blog/mui-core-v5.md

```diff
@@ -0,0 +1,683 @@
+
+<p class="blog-description">The popup explains what the <code>forcePopupIcon</code> prop is for.</p>
+
+These TypeScript prop descriptions are also used to generate the [API pages](/api/autocomplete/#props) of the documentation, so there is a single source of truth.
+
+### Migration from Enzyme to Testing Library
+
+The migration of MUI from class components [to hooks in v4](/blog/material-ui-v4-is-out/#preparing-for-the-future) broke many of the tests we had written with [Enzyme](https://github.com/enzymejs/enzyme/).
+Our tests were too coupled with the internals of React.
+We decided to transition our tests suite to [Testing Library](https://testing-library.com/).
+
+Fast forward a couple of years, Testing Library is now [the most popular](https://npm-stat.com/charts.html?package=enzyme&package=%40testing-library%2Freact&from=2020-09-13&to=2021-09-13) option to test React components in the community.
+
+We are happy to report that we have migrated **all** our test codebase to Testing Library.
+You can use our tests as inspiration when you need to write new ones.
+The migration has even allowed us to rethink some of the implementations, making the components easier to test with the library.
+
+### TypeScript migration
+
+The MUI Core codebase is not completely written in TypeScript yet but we have been coming a long way ([MUI X](https://github.com/mui-org/material-ui-x) is).
+In v4, we have written all the demos TypeScript first.
+
+With v5, we have made new steps toward the adoption of TypeScript:
+
+- We have made the TypeScript definitions the source of the API pages.
+  This reduces the likeliness for a new release to include out-of-date definitions.
+- We have migrated our first component to be written in TypeScript.
+- We are writing most of the new code in TypeScript.
+
+Overall, the language stats of GitHub give some qualitative measure of the progress:
+
+- 02/2019: 1.6%
+- 03/2020: 3.2%
+- 07/2020: 5.3%
+- 10/2020: 25.4%
+- 07/2021: 35.4%
+- Today: 36.1%
+
+<img src="/static/blog/mui-core-v5/typescript.png" alt="" style="width: 310px; margin-bottom: 16px;" />
+
+We expect the adoption of TypeScript to increase during the lifecycle of v5.
+With this organic adoption strategy, it might take us two years to close [#15984](https://github.com/mui-org/material-ui/issues/15984).
+
+### Strict Mode support
+
+This release is compatible with `React.StrictMode`.
+This is the result of two years of work on the infrastructure.
+We now run the tests and the documentation in strict mode.
+Thanks for the patience, this has been an old and [frequently requested](https://github.com/mui-org/material-ui/issues/13394) improvement.
+
+Bonus point, we run [React v18](https://github.com/reactwg/react-18) (unreleased) in a nightly build, in strict mode, without any test failures.
+
+> ⚠️ `@mui/styles` is not compatible with strict mode nor concurrent mode.
```

Comment:
```suggestion
> ⚠️ `@mui/styles` is not compatible with strict mode nor concurrent features.
```

The term "concurrent mode" is no longer applicable: https://github.com/reactwg/react-18/discussions/64